src/V3PerPackage/EnqueueCommand.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Services.Metadata.Catalog;
+using NuGet.Services.Metadata.Catalog.Persistence;
+
+namespace NuGet.Services.V3PerPackage
+{
+    public class EnqueueCommand
+    {
+        private readonly EnqueueCollector _collector;
+
+        public EnqueueCommand(EnqueueCollector collector)
+        {
+            _collector = collector;
+        }
+
+        public async Task ExecuteAsync(bool restart)
+        {
+            var fileSystemStorage = new FileStorageFactory(
+                new Uri("http://localhost/"),
+                Directory.GetCurrentDirectory());
+
+            var front = new DurableCursor(
```

Comment:
We use a local (file system) cursor just for enqueueing catalog leaves into a storage queue. When we run the real V3 components, a normal blob storage based cursor is used.